File: Java/Lanchonete/src/Aplicativo/Produto.java
package Aplicativo;

public class Produto {
    private final int codigo; //Codigo do produto
    private final String nome; //Nome do produto
    private final double preco; //Preco do produto
    private int quantidadeVendas; //Quantidade do produto que foi vendido

    public Produto(int codigo, String nome, double preco, int quantidadeVendas){
        this.codigo = codigo;
        this.nome = nome;
        this.preco = preco;
        this.quantidadeVendas = quantidadeVendas;
    }

    public Produto(int codigo, String nome, double preco){
        this(codigo, nome, preco, 0);
    }

    public double faturamento(){
        return preco * quantidadeVendas;
    }

    @Override
    public String toString() {
        return "Produto{" +
                "codigo=" + codigo +
                ", nome='" + nome + '\'' +
                ", preco=" + preco +
                ", quantidadeVendas=" + quantidadeVendas +
                '}';
    }

    public int getCodigo() {
        return codigo;
    }

    public String getNome() {
        return nome;
    }

    public double getPreco() {
        return preco;
    }

    public int getQuantidadeVendas() {
        return quantidadeVendas;
    }

    public void addQuantidadeVendas(int quantidade){
        quantidadeVendas += quantidade;
    }
}

File: C++/Lanchonete/Aplicativo/Produto.cpp
#include <iostream>
#include <string>
#include "Produto.h"

using namespace std;

Produto::Produto(int &codigo, string &nome, double &preco, int &quantidadeVendas) {
    this->codigo = codigo;
    this->nome = nome;
    this->preco = preco;
    this->quantidadeVendas = quantidadeVendas;
}

Produto::Produto(int &codigo, string &nome, double &preco) {
    this->codigo = codigo;
    this->nome = nome;
    this->preco = preco;
    this->quantidadeVendas = 0;
}

Produto::~Produto() = default;

double Produto::faturamento() const {
    return preco * quantidadeVendas;
}

int Produto::getCodigo() const {
    return codigo;
}

string Produto::getNome() {
    return nome;
}

double Produto::getPreco() const {
    return preco;
}

int Produto::getQuantidadeVendas() const {
    return quantidadeVendas;
}

void Produto::addQuantidadeVendas(int quantidade) {
    quantidadeVendas += quantidade;
}

ostream &operator<<(ostream &stream, const Produto &p) {
    stream << "Produto{" <<
         "codigo=" << p.codigo <<
         ", nome='" << p.nome << '\'' <<
         ", preco=" << p.preco <<
         ", quantidadeVendas=" << p.quantidadeVendas <<
         '}';

    return stream;
}
